applicationset/controllers/clustereventhandler.go

```diff
@@ -82,3 +95,42 @@ func (h *clusterSecretEventHandler) queueRelatedAppGenerators(q addRateLimitingI
 		}
 	}
 }
+
+// nestedGeneratorsHaveClusterGenerator iterate over provided nested generators to check if they have a cluster generator.
+func (h *clusterSecretEventHandler) nestedGeneratorsHaveClusterGenerator(generators []argoprojiov1alpha1.ApplicationSetNestedGenerator) bool {
+	for _, generator := range generators {
+		if h.nestedGeneratorHasClusterGenerator(generator) {
+			return true
+		}
+	}
+	return false
+}
+
+// nestedGeneratorHasClusterGenerator checks if the provided generator has a cluster generator.
+func (h *clusterSecretEventHandler) nestedGeneratorHasClusterGenerator(nested argoprojiov1alpha1.ApplicationSetNestedGenerator) bool {
```

Comment:
Instead of just logging, can we propagate the error up to `queueRelatedAppGenerators` and then log/ignore it there? Just would be nice to consolidate the places where errors are logged/ignored.